pkg/oci/oci.go

```diff
@@ -108,23 +109,29 @@ func GetGadgetImage(ctx context.Context, image string, authOpts *AuthOptions, pu
 		}
 	}
 
-	manifest, err := getImageManifestForArch(ctx, imageStore, image, authOpts)
+	manifest, err := getImageManifest(ctx, imageStore, image, authOpts)
 	if err != nil {
 		return nil, fmt.Errorf("getting arch manifest: %w", err)
 	}
 
-	prog, err := getEbpfProgramFromManifest(ctx, imageStore, manifest)
+	prog, err := getLayerFromManifest(ctx, imageStore, manifest, eBPFObjectMediaType)
 	if err != nil {
 		return nil, fmt.Errorf("getting ebpf program: %w", err)
 	}
 
+	wasm, err := getLayerFromManifest(ctx, imageStore, manifest, wasmObjectMediaType)
+	if err != nil {
+		return nil, fmt.Errorf("getting wasm program: %w", err)
+	}
+
```

Comment:
This layer is optional, it shouldn't fail if the layer is not found.

Reply:
I'll address this in https://github.com/inspektor-gadget/inspektor-gadget/pull/2355#discussion_r1448655497 first.